File: csharp/wordy/Wordy.cs
using System;

public static class Wordy
{
    const string start = "What is ";

    public static int Answer(string question)
    {
        if(!question.StartsWith(start))
            throw new ArgumentException();

        var elements = 
            question.Substring(start.Length)
            .Replace("multiplied by", "multipliedby")
            .Replace("divided by", "dividedby")
            .TrimEnd('?')
            .Split(' ');

        if (elements.Length % 2 != 1)
            throw new ArgumentException();

        int parse(string source) => 
            int.TryParse(source, out int parsed) 
            ? parsed : throw new ArgumentException();

        var result = parse(elements[0]);

        for(var i = 1; i < elements.Length; i += 2)
        {
            var op = elements[i];
            var num = parse(elements[i + 1]);
            result = ActOut(op, result, num);
        }

        return result;
    }

    private static int ActOut(string op, int result, int num)
    {
        switch(op)
        {
            case "plus": return result + num;
            case "minus": return result - num;
            case "multipliedby": return result * num;
            case "dividedby": return result / num;
            default: throw new ArgumentException();
        }
    }
}

File: go/wordy/wordy.go
package wordy

import (
	"strconv"
	"strings"
)

func Answer(question string) (int, bool) {
	if !strings.HasPrefix(question, "What is ") || !strings.HasSuffix(question, "?") {
		return 0, false
	}

	body := question[len("What is ") : len(question)-1]
	body = strings.ReplaceAll(body, "divided by", "divided_by")
	body = strings.ReplaceAll(body, "multiplied by", "multiplied_by")
	parts := strings.Split(body, " ")

	if len(parts) == 0 {
		return 0, false
	}

	result, err := strconv.Atoi(parts[0])
	if err != nil {
		return 0, false
	}
	parts = parts[1:]

	isOp := false
	op := func(a, b int) int { return 0 }
	for len(parts) > 0 {
		converted, err := strconv.Atoi(parts[0])
		if err == nil && isOp {
			result = op(result, converted)
			isOp = false
			parts = parts[1:]
			continue
		}

		if isOp {
			return 0, false
		}
		isOp = true

		switch parts[0] {
		case "plus":
			op = func(a, b int) int { return a + b }
		case "minus":
			op = func(a, b int) int { return a - b }
		case "divided_by":
			op = func(a, b int) int { return a / b }
		case "multiplied_by":
			op = func(a, b int) int { return a * b }
		default:
			return 0, false
		}

		parts = parts[1:]
	}

	if isOp {
		return 0, false
	}
	return result, true
}
